glTF/glTF-Mesh-Features-Samples-Sandcastle.js

```diff
@@ -0,0 +1,119 @@
+}
+
+// Creates the tileset from the tileset.json in the given subdirectory
+function createTileset(subdirectory) {
+  if (Cesium.defined(currentTileset)) {
+    viewer.scene.primitives.remove(currentTileset);
+    currentTileset = undefined;
+  }
+  // Create the tileset, and move it to a certain position on the globe
+  currentTileset = viewer.scene.primitives.add(
+    new Cesium.Cesium3DTileset({
+      url: `http://localhost:8003/${subdirectory}/tileset.json`,
+      debugShowBoundingVolume: true,
+    })
+  );
+  currentTileset.modelMatrix = Cesium.Transforms.eastNorthUpToFixedFrame(
+    Cesium.Cartesian3.fromDegrees(-75.152325, 39.94704, 0)
+  );
+  const offset = new Cesium.HeadingPitchRange(
+    0,
+    Cesium.Math.toRadians(-22.5),
+    4.0
+  );
+  viewer.zoomTo(currentTileset, offset);
+
+  // Assign the custom shader that visualizes the feature IDs to the tileset
+  currentTileset.customShader = createCustomShader();
+}
+
+//============================================================================
+// Sandcastle UI setup:
+
+// Create a label that will show some information for
+// the currently selected example
+function createInfoTextDisplay() {
+  const infoTextDisplay = document.createElement("div");
+  infoTextDisplay.style.background = "rgba(42, 42, 42, 0.7)";
+  infoTextDisplay.style.padding = "5px 10px";
+  infoTextDisplay.style.marginTop = "5px";
+  return infoTextDisplay;
+}
+const infoTextDisplay = createInfoTextDisplay();
+function setInfoText(infoText) {
+  infoTextDisplay.textContent = infoText;
+}
+
+// Create one entry for the list of examples that can
+// be selected in the dropdown menu. Selecting one of
+// these will load the tileset for the sample in the
+// given directory, and display the given info text in
+// the infoTexDisplay
+function createSampleOption(name, directory, infoText) {
+  return {
+    text: name,
+    onselect: function () {
+      createTileset(directory);
+      setInfoText(infoText);
+    },
+  };
+}
+
+// Create the list of available samples, and add them
+// to the sandcastle toolbar
+const sampleOptions = [
+  createSampleOption(
+    "FeatureIdAttribute",
+    "glTF/EXT_mesh_features/FeatureIdAttribute",
+    "Feature IDs for the vertices, using a feature ID attribute"
+  ),
+
+  createSampleOption(
+    "FeatureIdTexture",
+    "glTF/EXT_mesh_features/FeatureIdTexture",
+    "Feature IDs for texels, using a feature ID texture"
+  ),
+];
+Sandcastle.addToolbarMenu(sampleOptions);
+
+// Add a toggle button to the toolbar, for selecting whether
+// feature IDs should be visualized. When it is selected,
+// then the tileset receives the custom shader that visualizes
+// the feature IDs. Otherwise, the custom shader of the tileset
+// is set to 'undefined', causing it to be rendered with the
+// default shader.
+Sandcastle.addToggleButton("Visualize Feature IDs", true, function (checked) {
+  if (checked) {
+    currentTileset.customShader = createCustomShader();
```

Comment:
nitpick: constructing the shader every time is unnecessary. If you saved it in a global variable, you could reuse it

Reply:
In these small examples it's not a big enough difference to worry about, I'd say never mind about my suggestion here and in the other Sandcastle.